src/Sylius/Bundle/ApiBundle/Context/TokenBasedUserContext.php

```diff
@@ -30,9 +30,10 @@ public function getUser(): ?UserInterface
             return null;
         }
 
-        /** @var UserInterface|string $user */
+        /** @var UserInterface|string|null $user */
         $user = $token->getUser();
-        if (is_string($user)) {
+
+        if (is_string($user) || null === $user) {
             return null;
```

Comment:
Most of this method could be condensed into
```php
$user = $this->tokenStorage()->getToken()?->getUser();

return $user instanceof UserInterface ? $user : null;
```
I'm guessing we'll be only supporting PHP 8 and up when the support for symfony 6 is finished.